src/Role.php

```diff
@@ -8,6 +8,8 @@
 
 class Role extends Model
 {
+    protected $keyType = config('nova-permissions.role_model_key_type', 'int');
```

Comment:
This wouldn’t work as expected because model properties like `$keyType` are initialized during class compilation, while `config()` is a runtime function. So, using `config()` directly in the property declaration won't evaluate correctly.

Reply:
Yeah, I was able to finally figure this out and there is lots of things you have to do in order to make it running with UUID support (in my case also tenancy package and multiple databases). Unfortunately this is way outside of scope of any PR that can be merged directly into this project. Sorry for the fuss @pktharindu :)